File: vinzi.jsonMgt/src/main/clojure/vinzi/jsonMgt/hib/clj/errorEntry.clj
(ns vinzi.jsonMgt.hib.clj.errorEntry
    (:use vinzi.hib-connect.globals)
    (:import vinzi.jsonMgt.hib.errorEntry))

(defrecord errorEntry_clj [id datetime track command error user])

(defn create-errorEntry [datetime track command error user]
  ;; TO DO:  insert type-checking (java-objects are strictly typed)
  (errorEntry_clj. nil datetime track command error user))

(defn clone-errorEntry [cRec]
  ;; TO DO:  insert type-checking cRec should be same type/class as target
  (create-errorEntry (:datetime cRec) (:track cRec) (:command cRec) (:error cRec) (:user cRec)))

(defn errorEntry-to-java [cRec]
	(errorEntry. (:id cRec) (:datetime cRec) (:track cRec) (:command cRec) (:error cRec) (:user cRec) ))


(add-to-java (errorEntry_clj. 0 1 2 3 4 5) errorEntry-to-java)

(defn errorEntry-to-clj [jRec]
	(errorEntry_clj. (.getId jRec) (.getDatetime jRec) (.getTrack jRec) (.getCommand jRec) (.getError jRec) (.getUser jRec) ))


(add-to-clj (errorEntry.) errorEntry-to-clj)

File: vinzi.jsonMgt/src/main/java/vinzi/jsonMgt/hib/errorEntry.java
package vinzi.jsonMgt.hib;

public class errorEntry {
    public Long id;
    public Long getId() { return id;}
    protected void setId(Long id)  {this.id = id; };


    public java.util.Date datetime;
    public java.util.Date getDatetime() { return datetime;}
    public void setDatetime(java.util.Date datetime)  {this.datetime = datetime; };


    public String track;
    public String getTrack() { return track;}
    public void setTrack(String track)  {this.track = track; };


    public String command;
    public String getCommand() { return command;}
    public void setCommand(String command)  {this.command = command; };


    public String error;
    public String getError() { return error;}
    public void setError(String error)  {this.error = error; };


    public String d_user;
    public String getD_user() { return d_user;}
    public void setD_user(String d_user)  {this.d_user = d_user; };


    public errorEntry() {};

    public errorEntry(Long id, java.util.Date datetime, String track, String command, String error, String d_user) {
	this.id = id;
	this.datetime = datetime;
	this.track = track;
	this.command = command;
	this.error = error;
	this.d_user = d_user;
	return;};

   public String toString() {
        return " id=" + this.id + " datetime=" + this.datetime + " track=" + this.track + " command=" + this.command + " error=" + this.error + " d_user=" + this.d_user;};

};
